### ISSUE TITLE
Postman Collection Runner - not clearing cookies during run

### ISSUE THREAD

**Issue by rufreakde** [a85/POSTMan-Chrome-Extension#9675]
**Describe the Issue**
using `jar.clear()` during a postman runner run will not clear the cookies.

So I have two folders and each of the folders has a request with a test section to delete all the cookies I use and check if they are really deleted in the test. (This code is tested and seems to work when run in postman but fails in a postman collection runner)

So when I run folder 1 in the runner and afterwards folder two (the order in which they would usually run) they succeed. But when I run both at the same time they fail. So I did some testing and executed only the first folder with the clear request of the second folder and the cookies persist in postman! (I will add some screenshots for understanding) 

Is there a workaround for this issue I need both executions in one solution to be able to send requests to two 2 destinations that use a SAML authentication flow.

**Steps to Reproduce**
* Given two folders with requests
<img width="281" alt="Screenshot 2021-03-01 at 20 55 51" src="https://user-images.githubusercontent.com/28104368/109551349-bd246f80-7ad0-11eb-81e2-da5388575e4c.png">
* Given cookie clear request in each folder as first request

```js
pm.test(`CLEAR COOKIES`, () => {
    // clearing all the cookies from these urls
    // needs to whitelist with '*'
    const jar = pm.cookies.jar();

    const auth_idp = pm.globals.get("auth_idp")
    const auth_idp_url = `https://${auth_idp}`;

    jar.clear(auth_idp_url, (error) => {
        if (error){
            console.error(auth_idp_url);
            console.error(error);
        } else {
            jar.getAll(auth_idp_url, (error, cookies) => {
                if (error){
                    console.error(auth_idp_url);
                    console.error(error);
                }
                pm.expect(cookies.length).to.be.undefined;
            });
        }
    });
    ...
    console.log("CLEARED COOKIES");
});
```

* run postman collection
<img width="1018" alt="Screenshot 2021-03-01 at 21 01 27" src="https://user-images.githubusercontent.com/28104368/109552005-90bd2300-7ad1-11eb-968e-4c1f76d96365.png">
* see that cookies still persist (also my second folder collection fails because of cookies)
<img width="709" alt="Screenshot 2021-03-01 at 21 04 45" src="https://user-images.githubusercontent.com/28104368/109552314-e5609e00-7ad1-11eb-9268-f8f42bb307db.png">
* run only the Clear Cookies request via postman and all cookies are cleared
<img width="714" alt="Screenshot 2021-03-01 at 21 07 40" src="https://user-images.githubusercontent.com/28104368/109552604-4ab48f00-7ad2-11eb-96d8-60e80504559a.png">

**Environment Information**
 - Platform Type: 
 - Postman Version: Version 7.36.5 (7.36.5)
 - Operating System: MAC OS Big Sur 11.2.1

**Additional Context**
It seems the postman collections runner is not executing `jar.clear()` at all! Not at the start nor in the end of a run.
And it fails silently.

Another thing I noticed:
If jar.clear gets called with a URL that does not exist it fails silently! And even worse all subsequent jar.clear() calls fail then as well!

**Comment by DannyDainton** [comment]
Hey @rufreakde

As I wasn't able to see this in the information (mentioned in the script comment but couldn't see if it was done) you shared, can you confirm that the domains have been whitelisted?

Could you share an image from the `Whitelist Domains` section? 

Just want to rule that out before we investigate this from our side.

**Comment by rufreakde** [comment]
@DannyDainton Hi Danny yes, otherwise I would not be able to delete the cookies when the request is executed manually.

I also added * in the whitelist wich also seems to work now for all pages. Is this not intended behavior? If so will add all routes specifically in addition tomorrow just in case.

**Comment by DannyDainton** [comment]
Thanks for the response 🏆

I did remember an issue a while back where it didn't appear to work as this particular user didn't also add the protocol to the whitelisted domain value:

https://github.com/postmanlabs/postman-app-support/issues/7977#issuecomment-600052882

Not saying that's what's happening here but worth just confirming.

**Comment by rufreakde** [comment]
@DannyDainton I removed my wildcard * and added all domains specifically.
I rechecked manually if the cookies are deleted with the requests and it still works.

Again running in Collection Runner results in the same issue. Not clearing the Cookies correctly.

The strange thing is that the actual Clear Request Test for succeeds.
<img width="1022" alt="Screenshot 2021-03-02 at 08 16 32" src="https://user-images.githubusercontent.com/28104368/109615275-022dbd80-7b34-11eb-83f0-5d8ce984a279.png">


So I rechecked and made the test code a little bit more robust and added logging:
```js
const jar = pm.cookies.jar();

    const auth_idp = pm.globals.get("auth_idp")
    const auth_idp_url = `https://${auth_idp}`;
    jar.clear(auth_idp_url, (error) => {
        if (error){
            console.error(auth_idp_url);
            console.error(error);
        } else {
            jar.getAll(auth_idp_url, (error, cookies) => {
                if (error){
                    console.error(auth_idp_url);
                    console.error(error);
                }
                const cookies_string = JSON.stringify(cookies);
                console.log(`cookies: ${auth_idp_url}`, cookies_string);
                pm.expect(cookies_string).to.be.eq("[]");
            });
        }
    });
```
Now I saw something strange the second Clear Cookies request (which is the same code as the first request) is not clearing all the cookies! (missing console log statements) 
1. request works in runner as intended:
<img width="679" alt="Screenshot 2021-03-02 at 08 39 00" src="https://user-images.githubusercontent.com/28104368/109614629-29d05600-7b33-11eb-922f-8b4a928c6971.png">
2. request is only printing one:
<img width="624" alt="Screenshot 2021-03-02 at 08 42 57" src="https://user-images.githubusercontent.com/28104368/109614879-7f0c6780-7b33-11eb-9582-deb2271a2176.png">
Failed cookie jar clear without any error message. Looks like a bug.

**Comment by rufreakde** [comment]
For anyone else also encountering this issue I found a **workaround**:
Instead of clearing all the jars in one request I just created a request for each jar (it is tiresome to execute but well I use a runner for it)
<img width="287" alt="Screenshot 2021-03-02 at 09 53 34" src="https://user-images.githubusercontent.com/28104368/109623409-699c3b00-7b3d-11eb-858d-164b30283596.png">
So now I have a folder with all the "cleanup" one by one.

**Comment by mohitranka** [comment]
Hi @rufreakde ! This issue has been fixed in the latest [web version](web.postman.co). It will be available in the next major app version ( 8.0.7 ) when released, as well.

**Comment by rufreakde** [comment]
@mohitranka I updated now to version 8
<img width="272" alt="Screenshot 2021-03-11 at 08 42 25" src="https://user-images.githubusercontent.com/28104368/110752644-d605f100-8245-11eb-8c5d-79833a80f370.png">

And it seems it made it worse. Now not even my workaround works anymore... same tests run and on clear cookies they are just still there. Executing the clear manually clears them....

Is there maybe something like "CookieJar.DeleteAllCookies()" (not per route) option that works all the time for repetitive changes?

**Comment by rufreakde** [comment]
Workaround:
In the end I split the collection in two and now it works somehow. (no runner code or other changes just moved in seperate collections)

**Comment by mccannt** [comment]
Issue has been resolved in v8.0.7 as well as web. 
Closing issue.
